packages/springboard/src/components/BaseModal.vue

```diff
@@ -0,0 +1,35 @@
+<template>
+  <div class="overlay" ref="overlay" @click.self="$emit('close')" role="dialog">
+    <div class="wrapper">
+      <CloseButton @click="$emit('close')"></CloseButton>
+      <div class="content">
+        <slot></slot>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+import CloseButton from "./CloseButton.vue";
+
+export default defineComponent({
+  components: {
+    CloseButton,
+  },
+  emits: {
+    close: null,
+  },
+});
+</script>
+
+
+<style lang="scss" scoped>
+.overlay {
+  @apply bg-gray-900 bg-opacity-40 absolute block top-0 left-0 w-[calc(100%)] h-[calc(100%)];
```

Comment:
Since you use it only once, that combination should not be an apply.
It should be inline classes.

Reply:
Why? I like the cleaner template syntax.